vic/CMakeLists.txt

```diff
@@ -0,0 +1,207 @@
+
+set(CMAKE_INSTALL_PREFIX /usr/local/vic) # Path to install VIC shared library.
+
+set(CMAKE_C_COMPILER gcc)
+set(CMAKE_C_FLAGS_RELEASE ${CMAKE_C_FLAGS_RELEASE} )
+set(CMAKE_C_FLAGS_DEBUG  ${CMAKE_C_FLAGS_DEBUG} )
+
+execute_process(COMMAND git describe --abbrev=4 --dirty --always --tags OUTPUT_VARIABLE GIT_VERSION)
+execute_process(COMMAND uname -s OUTPUT_VARIABLE UNAME_S)
+execute_process(COMMAND uname -n OUTPUT_VARIABLE HOSTNAME)
+
+message(STATUS "GIT_VERSION: ${GIT_VERSION}")
+message(STATUS "UNAME_S: ${UNAME_S}")
+message(STATUS "HOSTNAME: ${HOSTNAME}")
+
+include_directories(./vic_run/include ./drivers/shared_all/include
+        ./drivers/image/include ./drivers/shared_image/include
+        ./drivers/classic/include ${OTHER_INCLUDE_PATH})
+link_directories(${OTHER_LIBRARY_PATH})
+
+file(GLOB VIC_RUN_SRC ./vic_run/src/*.c)
+file(GLOB VIC_CLASSIC_SRC ./drivers/classic/src/*.c)
+file(GLOB VIC_IMAGE_SRC ./drivers/image/src/*.c)
+
+file(GLOB VIC_RUN_HDR ./vic_run/include/*.h)
+file(GLOB VIC_CLASSIC_HDR ./drivers/classic/include/*.h)
+file(GLOB VIC_IMAGE_HDR ./drivers/image/include/*.h)
+
+file(GLOB VIC_SHARE_SRC ./drivers/shared_all/src/*.c)
+file(GLOB VIC_SHARE_HDR ./drivers/shared_all/include/*.h)
+
+file(GLOB VIC_SHARE_IMG_SRC ./drivers/shared_image/src/*.c)
+file(GLOB VIC_SHARE_IMG_HDR ./drivers/shared_image/include/*.h)
+
+# ----------------------------------------------------------------------------
+# VIC Classic Driver
+# ----------------------------------------------------------------------------
+if(CLASSIC)
+    set(CLASSIC_SRC ${VIC_CLASSIC_SRC} ${VIC_SHARE_SRC} ${VIC_RUN_SRC}
+                    ${VIC_CLASSIC_HDR} ${VIC_SHARE_HDR} ${VIC_RUN_HDR})
+
+    add_executable(vic_classic ${CLASSIC_SRC})
+    target_link_libraries(vic_classic m)
+endif()
+
+# ----------------------------------------------------------------------------
+# VIC Image Driver
+# ----------------------------------------------------------------------------
+
+#           Find out the include and library directory of MPI.
+if(IMAGE)
+
+    # Probably include and library paths of MPI.
+     set(MPI_LIB_SEARCH_PATHS
+            $ENV{HOME}/mpich/lib
+            $ENV{HOME}/openmpi/lib
+            /lib/
+            /lib/openmpi
+            /lib64/
+            /usr/lib
+            /usr/lib/openmpi
+            /usr/lib/x86_64-linux-gnu
+            /usr/lib64
+            /usr/local/lib
+            /usr/local/lib64
+            /usr/local/openmpi/lib
+       )
+    set(MPI_INCLUDE_SEARCH_PATHS
+            /usr/include
+            /usr/local/include
+       )
+    set(MPICH_INCLUDE_SEARCH_PATHS
+            $ENV{HOME}/mpich/include
+            /usr/include/mpich
+            /usr/local/include/mpich
+            /usr/local/mpich/include
+       )
+    set(OPENMPI_INCLUDE_SEARCH_PATHS
+            $ENV{HOME}/openmpi/include
+            /usr/include/openmpi
+            /usr/local/include/openmpi
+            /usr/local/openmpi/include
+       )
```

Comment:
I don't think we need any of these hard coded paths. The `mpicc` compiler should include all include and linking flags necessary to compile VIC.